internal/controller/backup_controller.go

```diff
@@ -560,6 +567,10 @@ func startInstanceManagerBackup(
 	pod *corev1.Pod,
 	cluster *apiv1.Cluster,
 ) error {
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return fmt.Errorf("pod container statuses list is empty")
+	}
```

Comment:
Here I'm not convinced.
If we emit this error, the calling function (`Reconcile`) will immediately mark the backup as failed, with no retry.
Whereas, a few lines previous, in a check for `utils.IsPodReady(*pod)` if it is not ready, it will retry.
I think retry is better.
Maybe we can take this as some sort of retriable condition.

Reply:
Yeah, it could be retry is better, I have no strong opinion. 
We should decide and go for the behaviour we want.

Reply:
Added the retry logic, please have a look to make sure i didn't go crazy